### PR TITLE
Unified component locale support

### PR DESCRIPTION
**Pull request by afc163**
ref #1055 #1042

### CODE REVIEW

File: components/date-picker/PickerMixin.jsx [hunk 1]
```diff
@@ -1,14 +1,23 @@
+import React from 'react';
 import objectAssign from 'object-assign';
 import defaultLocale from './locale/zh_CN';
 import DateTimeFormat from 'gregorian-calendar-format';
 import GregorianCalendar from 'gregorian-calendar';
 
 export default {
+  contextTypes: {
+    locale: React.PropTypes.object,
```
Comment (yiminghe) [inline components/date-picker/PickerMixin.jsx]: 改名为 antdLocale ，太容易和其他库冲突

Comment (plandem) [inline components/date-picker/PickerMixin.jsx]: please update demo for react-intl too:
http://yiminghe.me/learning-react/examples/react-intl.html?locale=en-US

File: components/modal/confirm.jsx [hunk 2]
```diff
@@ -5,8 +5,24 @@ import Icon from '../icon';
...
+  justOkText: '知道了',
+};
+
+let runtimeLocale = { ...defaultLocale };
+
+export function changeConfirmLocale(newLocale) {
+  if (newLocale) {
+    objectAssign(runtimeLocale, newLocale);
+  } else {
+    runtimeLocale = defaultLocale;
```
Comment (benjycui) [inline components/modal/confirm.jsx]: `runtimeLocale = defaultLocale;` => `runtimeLocale = {...defaultLocale};`

不然 英文 -> 中文 -> 英文，`defaultLocale` 就被覆盖了。

Comment (benjycui) [inline components/modal/confirm.jsx]: fix in: https://github.com/ant-design/ant-design/commit/b8d040f9438c464b02f1c70b1e6ebf636eb3cab2

**Comment by benjycui** [comment]
Rebased

**Pull request merged**